bot/exts/events/advent_of_code/_cog.py

```diff
@@ -52,13 +60,80 @@ def __init__(self, bot: Bot):
         self.status_task.set_name("AoC Status Countdown")
         self.status_task.add_done_callback(_helpers.background_task_callback)
 
+        self.completer_task.start()
+
+    @tasks.loop(minutes=10.0)
+    async def completer_task(self) -> None:
+        """
+        Synchronise the members with the completionist role and members who has completed all 50 AoC stars.
+
+        Adds the role to those that need it, removes the role from those who have been added ot the block list.
+        Runs on a schedule, as defined in the task.loop decorator.
+        """
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(Client.guild)
+        completionist_role = guild.get_role(Roles.aoc_completionist)
+
+        aoc_name_to_member_id = {
+            aoc_name: member_id
+            for member_id, aoc_name in await self.account_links.items()
+        }
+
+        try:
+            leaderboard = await _helpers.fetch_leaderboard()
+        except _helpers.FetchingLeaderboardFailedError:
+            await self.bot.send_log("Unable to fetch AoC leaderboard during role sync.")
+            return
+
+        placement_leaderboard = json.loads(leaderboard["placement_leaderboard"])
+        members_who_deserve_role: set[discord.Member] = set()
+
+        for member_aoc_info in placement_leaderboard.values():
+            if not member_aoc_info["stars"] == 50:
+                # Only give the role to people who have completed all 50 stars
+                continue
+
+            member_id = aoc_name_to_member_id.get(member_aoc_info["name"], None)
+            if not member_id:
+                continue
+
+            member = await members.get_or_fetch_member(guild, member_id)
+            if member is None:
+                continue
+
+            if member in completionist_role.members:
+                if await self.completionist_block_list.contains(member_id):
+                    await members.handle_role_change(member, member.remove_roles, completionist_role)
```

Comment:
Accidentally deleted the comment, so here it is again:

The block command is the only place where `completionist_block_list` is mutated. That command also removes the role before mutating. Thus, in what case could a blocked user still have their role? This role change seems redundant.

Reply:
Yea, good point. Removed in https://github.com/python-discord/sir-lancebot/pull/991/commits/2fe89228fd3e583f2c8746d7df08af9c1c234972